library/dd-sdk-android-session-replay/src/main/kotlin/com/datadog/android/sessionreplay/internal/recorder/mapper/UnsupportedViewMapper.kt

```diff
@@ -0,0 +1,88 @@
+/*
+ * Unless explicitly stated otherwise all files in this repository are licensed under the Apache License Version 2.0.
+ * This product includes software developed at Datadog (https://www.datadoghq.com/).
+ * Copyright 2016-Present Datadog, Inc.
+ */
+
+package com.datadog.android.sessionreplay.internal.recorder.mapper
+
+import android.view.View
+import com.datadog.android.sessionreplay.internal.recorder.MappingContext
+import com.datadog.android.sessionreplay.model.MobileSegment
+
+internal class UnsupportedViewMapper :
+    BaseWireframeMapper<View, MobileSegment.Wireframe.TextWireframe>() {
+
+    override fun map(view: View, mappingContext: MappingContext):
+        List<MobileSegment.Wireframe.TextWireframe> {
+        val viewGlobalBounds = resolveViewGlobalBounds(
+            view,
+            mappingContext.systemInformation.screenDensity
+        )
+
+        return listOf(
+            MobileSegment.Wireframe.TextWireframe(
+                id = resolveViewId(view),
+                x = viewGlobalBounds.x,
+                y = viewGlobalBounds.y,
+                width = viewGlobalBounds.width,
+                height = viewGlobalBounds.height,
+                shapeStyle = resolveShapeStyle(view),
+                border = resolveBorder(),
+                text = view::class.java.name,
```

Comment:
so instead of the screenshot we are going to show the box with class name? What will be a solution for the case when class name is obfuscated?

Reply:
well this is only used right now explicitly for system classes so in theory those will not be obfuscated.

Reply:
cannot we have this class to be used with say views coming from `androidx` or any other non-system package?

Reply:
Changed to have a static text - for toolbars this will be "Toolbar", and as a fallback for unsupported views that are not explicitly mapped it will be "Unsupported view"